dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java

```diff
@@ -646,6 +646,17 @@ public Period createPeriod( String isoDate )
     // CalendarPeriodType
     // -------------------------------------------------------------------------
 
+    public final Period getFuturePeriod( int periodOffset )
+    {
+        Period period = createPeriod();
+
+        // Rewind one as 0 open periods implies current period is locked
+
+        period = getPreviousPeriod( period );
+
+        return getNextPeriod( period, periodOffset );
```

Comment:
We talked about this logic. I see the need for it now. If periodOffset is 0, then this returns the previous period. If you just replace these two lines with `return getNextPeriod( period, periodOffset - 1 )` that would return the current period when periodOffset is 0 because -1 results in no shift within `getNextPeriod`.

However you could replace these two lines with `return getShiftedPeriod( period, periodOffset - 1 )`, though it won't backport unless you backport `getShiftedPeriod` too -- I just added it recently to work with yearToDate and periodOffset stuff, thinking it might be useful for other things some day!

Reply:
Thanks for providing this. For now I just keep it the way it is.